### ISSUE TITLE
Fail if an user other than "prisma" is used in the client code generation

### ISSUE THREAD

**Issue by davidmartos96** [electric-sql/electric#701]
Hello!
I was wondering how good of an idea would be to fail the CLI code generation if the `proxy` param sends a proxy URL with an user different than "prisma". Say the CLI user sends the regular postgres connection URL.
What that would do is generate an schema including non electrified tables. Since the "prisma" user has a special behavior in the introspection, it might be a good idea to check so.

Context: https://discord.com/channels/933657521581858818/1178328362532491397/1178555231513100318

Looking at the generated code, a table called `schema_migrations` has been generated. Possibly coming from a postgres migrations CLI utility like `go-migrate`.
And looking at the migrations file, only `todolist` appears to be electrified. My guess is that this user electrified only the `todolist` table and then used the CLI with the regular postgres connection URL. That introspected all tables for the schema file, and only `todolist` for the migrations because it comes separately from the API.

**Comment by kevin-dp** [comment]
Hi @davidmartos96, adding extra barriers to prevent unexpected usage surely sounds like a good idea. May save some time spent in debugging.

**Comment by samwillis** [comment]
Hey @davidmartos96, I have been working though a large refactor of our client side CLI tools, this is addressed in this PR #664 and will land in a future release.

If the user passes a --proxy argument, its is deconstructed into parts and used to construct the correct pg-path for the generator:

https://github.com/electric-sql/electric/blob/61ba459fe3b1a607c6a9d85cbb2890f5cb38d2c3/clients/typescript/src/cli/migrations/migrate.ts#L262-L283

We are aiming for users to use .env files, and so if it is:
```
ELECTRIC_SERVICE_HOST=my-electric-host
ELECTRIC_HTTP_PORT=5133
ELECTRIC_DATABASE_URL=postgresql://something:something@host:5432/postgres
ELECTRIC_PG_PROXY_PORT=65432
ELECTRIC_PG_PROXY_PASSWORD=password
```
Everything will just work without having to manually configure the generator with arguments

The related starter PR #676 changes the client:generate command to be just:
```
"client:generate": "npx electric-sql generate",
```

**Comment by davidmartos96** [comment]
@samwillis Thank you! That looks easier
